### DISCUSSION TITLE
TransformScene Issue

### DISCUSSION THREAD

**Discussion by mattwyliearo** [cdcseacave/openMVS#1095] [Q&A]
Hello! I'm working on a project right now and I'm attempting to use the TranformScene feature, but I'm running into an issue where there seems to be an issue with the arguments. My goal is to set the up-axis for the scene to y using the "--up-axis 2" argument, but it keeps outputting the Available options and fails to update the mvs file. Can you identify what I'm doing incorrectly?

:: Convert COLMAP's output to OpenMVS's input
%oMVS%\InterfaceColmap.exe -i %workDir%\myundist -o %workDir%\proj.mvs

:: Continue with OpenMVS processing
cd myundist
%oMVS%\DensifyPointCloud.exe %workDir%\proj.mvs

:: After mesh reconstruction, before mesh refinement
%oMVS%\ReconstructMesh.exe %workDir%\proj_dense.mvs

:: Set the up-axis for the scene
%oMVS%\TransformScene.exe -i %workDir%\proj_dense.mvs -o %workDir%\proj_dense_transformed.mvs --up-axis 1

%oMVS%\RefineMesh.exe --resolution-level 1 %workDir%\proj_dense_transformed.mvs
%oMVS%\TextureMesh.exe --export-type obj -o %outputFileName%.obj %workDir%\proj_dense_mesh_refine.mvs

**Comment by cdcseacave** [comment]
can you pls write a python script for the commands you are using, and a small dataset where I can reproduce the problem?

**Reply by mattwyliearo** [reply]
Here's the script. I'm zipping up the files and folder structure to make it a quick open and run (includes the COLMAP proj.mvs file in workDir and corresponding images for the proj.mvs file). I'm using oMVS v2.2.0 binaries (already in the zipped file) as 2.3.0 has been giving me issues.

import subprocess
import os


def ensure_dir(directory):
    """Ensure directory exists."""
    if not os.path.exists(directory):
        os.makedirs(directory)


def construct_command(executable, args):
    """Construct command for subprocess."""
    return f"{executable} {' '.join(args)}"


def main():
    # Define the paths
    scriptDir = os.path.dirname(os.path.abspath(__file__))
    oMVS = os.path.join(scriptDir, "oMVS")
    workDir = os.path.join(scriptDir, "workDir")
    modelsDir = os.path.join(scriptDir, "models")

    # Ensure the models directory exists
    ensure_dir(modelsDir)

    # Define file paths
    projMvsFile = os.path.join(workDir, "proj.mvs")
    outputFileName = "output_model"

    # Define the OpenMVS commands
    commands = [
        construct_command(f"{oMVS}\\DensifyPointCloud.exe", [projMvsFile]),
        construct_command(
            f"{oMVS}\\ReconstructMesh.exe",
            ["-i", os.path.join(workDir, "proj_dense.mvs")],
        ),
        construct_command(
            f"{oMVS}\\TransformScene.exe",
            [
                "--input-file",
                os.path.join(workDir, "proj_dense_mesh.mvs"),
                "--output-file",
                os.path.join(workDir, "proj_dense_mesh_transformed.mvs"),
                "--up-axis",
                "1",
            ],
        ),
        construct_command(
            f"{oMVS}\\RefineMesh.exe",
            [
                os.path.join(workDir, "proj_dense_mesh_transformed.mvs"),
                "-o",
                os.path.join(workDir, "proj_dense_mesh_refine.mvs"),
                "--scales",
                "1",
                "--max-face-area",
                "16",
            ],
        ),
        construct_command(
            f"{oMVS}\\TextureMesh.exe",
            [
                "--input-file",
                os.path.join(workDir, "proj_dense_mesh_refine.mvs"),
                "--output-file",
                os.path.join(modelsDir, f"{outputFileName}.obj"),
                "--export-type",
                "obj",
            ],
        ),
    ]

    # Execute the OpenMVS commands
    for cmd in commands:
        print(f"Executing: {cmd}")
        subprocess.run(cmd, shell=True)

    print("OpenMVS processing completed.")


if __name__ == "__main__":

    main()

[test_omvs.zip](https://github.com/cdcseacave/openMVS/files/14076841/test_omvs.zip)

**Comment by cdcseacave** [comment]
you sent me 3 bat files, where are the iamges, the mvs project, the python script, etc?

**Reply by mattwyliearo** [reply]
Sorry about that. I've updated the zip file above. Since MVS zipped is > 26MB, you'll need to extract oMVS 2.2.0 to the oMVS folder located.

**Reply by cdcseacave** [reply]
same file with nothing :)

**Reply by mattwyliearo** [reply]
Third time's the charm! Should be updated correctly now lol. It's been a long week!
